vulnerabilities/templates/packages.html

```diff
@@ -1,64 +1,202 @@
+                    <th style="width: 225px;"><span class="has-tooltip-multiline has-tooltip-black has-tooltip-arrow has-tooltip-text-left" data-tooltip="This is the number of vulnerabilities fixed by the package.">Fixing Vulnerabilities</span></th>
+                </tr>
+            </thead>
+            <tbody>
+                {% for package in packages %}
+                <tr>
+                    <td style="word-break: break-all;">
+                        <a href="{% url 'package_view' package.pk %}?type={{package_type}}&name={{package.package_url}}" target="_self">{{package.package_url}}</a>
+                    </td>
+                    <td>{{package.vulnerability_count}}</td>
+                    <td>{{package.patched_vulnerability_count}}</td>
+                </tr>
+                {% endfor %}
+            </tbody>
+        </table>
+    </div>
+
+    <nav class="pagination is-centered is-small" aria-label="pagination">
+        {% if packages.has_previous %}
+        <a href="?{{ searched_for }}&page={{ packages.previous_page_number }}" class="pagination-previous">Previous</a>
+        {% else %}
+        <a class="pagination-previous" disabled>Previous</a>
+        {% endif %}
+
+        {% if packages.has_next %}
+        <a href="?{{ searched_for }}&page={{ packages.next_page_number }}" class="pagination-next">Next</a>
+        {% else %}
+        <a class="pagination-next" disabled>Next</a>
+        {% endif %}
+
+        <ul class="pagination-list">
+            {% if packages.number != 1%}
+            <li>
+                <a href="?{{ searched_for }}&page=1{% if url_params_without_page %}&{{ url_params_without_page }}{% endif %}" class="pagination-link" aria-label="Goto page 1">1</a>
+            </li>
+            {% if packages.number > 2 %}
+            <li>
+                <span class="pagination-ellipsis">&hellip;</span>
+            </li>
+            {% endif %}
+            {% endif %}
+            <li>
+                <a class="pagination-link is-current" disabled aria-label="Page {{ packages.number }}" aria-current="page">{{ packages.number|intcomma }}</a>
+            </li>
+            {% if packages.number != packages.paginator.num_pages %}
+            {% if packages.next_page_number != packages.paginator.num_pages %}
+            <li>
+                <span class="pagination-ellipsis">&hellip;</span>
+            </li>
+            {% endif %}
+            <li>
+                <a href="?{{ searched_for }}&page={{ packages.paginator.num_pages }}{% if url_params_without_page %}&{{ url_params_without_page }}{% endif %}" class="pagination-link" aria-label="Goto page {{ packages.paginator.num_pages }}">{{ packages.paginator.num_pages|intcomma }}</a>
+            </li>
+            {% endif %}
```

Comment:
Use indentation after tag such as `if` and `for` to make the code easier to read, especially here where you have imbricated conditions.
See some examples at https://docs.djangoproject.com/en/4.0/ref/templates/language/#tags

Reply:
@tdruez I've started applying the indentation to the Jinja if/for tags, only to notice that when I save the file, some "feature" of VSCode is returning all HTML to the indented format structure one would expect if there were no Jinja templates.  Time for some troubleshooting....

Reply:
I've installed the Django extension from Baptiste Darthenay and now I can save Jinja tag indents, but my HTML intellisense/auo-complete has disappeared.

Reply:
Finished properly indenting all Jinja2 if/for tags.  VSCode problem has been widely discussed on StackOverflow and elsewhere for several years, but I've found no "solutions" that (1) enable the user to save properly-indented Jinja tags -and- (2) provide the usual VSCode HTML intellisense and auto-complete.  (1) is clearly necessary, and perhaps someday (2) will also be available at the same time.  🙂